File: playground/BOJ/boj_28074(s)/boj_28074.js
/*
----------------------------------------------
Sub : [BOJ] 모비스
Link: https://www.acmicpc.net/problem/28074
Tag : JS, String
Memo
----------------------------------------------
*/

const fs = require('fs');

const input = fs.readFileSync('/dev/stdin', 'utf8').toString().trim();

function canFormMobis(string) {
    const requiredChars = "MOBIS";
    for (const char of requiredChars) {
        if (!string.includes(char)) {
            return "NO";
        }
    }
    return "YES";
}

// 결과 출력
console.log(canFormMobis(input));

File: playground/BOJ/S_boj_28074/boj_28074.py
'''
----------------------------------------------
Sub : [BOJ] 모비스
Link: https://www.acmicpc.net/problem/28074
Tag : Python, String
Memo
----------------------------------------------
'''

def can_form_mobis(string):
    required_chars = "MOBIS"

    for char in required_chars:
        if string.count(char) < 1:
            return "NO"
    return "YES"

input_string = input().strip()

# 결과 출력
print(can_form_mobis(input_string))
